app/controllers/session_controller.rb

```diff
@@ -674,4 +800,31 @@ def redeem_invitation(invite, sso)
     Rails.logger.warn("SSO invite redemption failed: #{e}")
     raise Invite::RedemptionFailed
   end
+
+  def find_second_factor_challenge(nonce)
```

Comment:
Will do. 

> checks if the challenge was successful, not sure if that should also be here

No it shouldn't do that here because the challenge at this stage is not successful, but will be marked successful if the 2fa succeeds.